vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java

```diff
@@ -348,10 +337,10 @@ protected void onAttach(AttachEvent attachEvent) {
         if (locale == null) {
             getUI().ifPresent(ui -> setLocale(ui.getLocale()));
         } else if (languageTag != null) {
-            setLocaleWithJS();
+            requestI18nUpdate();
         }
         if (i18n != null) {
-            setI18nWithJS();
+            requestI18nUpdate();
```

Comment:
`setLocale` will implicitely call `requestI18nUpdate`, so there is a chance here that it will be called twice in a row.

However I think that's preferrable to making assumptions here about which method will implicitely call another method. To cope with multiple calls, I implemented `requestI18nUpdate` so that pending executions will be canceled.